### DISCUSSION TITLE
YoutubeDataAPI - Error 403 Forbidden when get youtube.members.list

### DISCUSSION THREAD

**Discussion by victorchiq** [googleapis/googleapis#898] [Q&A]
I am trying to use the endpoint /youtube/v3/members from YoutubeDataAPI, but even after the successful channel authorization with OAuth2, i keep getting the error:

{
    "error": {
        "code": 403,
        "message": "Access forbidden. The request may not be properly authorized.",
        "errors": [
            {
                "message": "Access forbidden. The request may not be properly authorized.",
                "domain": "youtube.common",
                "reason": "forbidden"
            }
        ]
    }
}

The authentication scopes that i used were: "https://www.googleapis.com/auth/youtube.channel-memberships.creator", "https://www.googleapis.com/auth/youtube.readonly"

I get the access_token, but when i call the API (direct in node.js with 'googleapis' youtube.members.list; or with Postman), i still get the 403 error. And yes, the API is enabled in my console.cloud, and my Oauth credential ID is WEB APP.

I tried the same access token that I used to call the /members endpoint for other endpoints an it worked successfully.

How can i solve this? Is there any more settings in the channel that needs to be changed?

**Comment by ohmiler** [comment]
Did you find a solution? I have the same problem.

**Comment by jvmMachado** [comment]
Have anyone found any solution? I get the same problem. The strange part is that through https://developers.google.com/ I can get the token and make the request with the expected response. But when I try it through https://developers.google.com/oauthplayground/ I get the same "Access forbidden" problem. But both oauth clients are managed by google's developers portal, so I don't get what is happening here. Scopes are right and everything.

**Comment by RobConings** [comment]
I'm also bouncing on this issue and every forum post about it on stackexchange or on git is just a dead thread

**Comment by Bumboobee** [comment]
### ✅ Definitive Answer: There’s No Practical Solution for This via Public API

We face the **exact same problem**, and after **months of exhaustive research**, I can confidently say:

> **There is NO practical solution available to programmatically retrieve YouTube channel members via the public API.**

Let me walk you through everything we’ve uncovered over the past **4+ months**, step by step:

---

### 1. Setting Up Scopes in Google Console

🔗 [[Google Cloud Console](https://console.cloud.google.com/)](https://console.cloud.google.com/)

If you're using **confidential scopes** such as:

* `https://www.googleapis.com/auth/youtube.readonly`
* `https://www.googleapis.com/auth/youtube.channel-memberships.creator`

…then you **must** submit your application for **OAuth verification** by Google’s security team.

### ✅ Key points:

* This process can take **2-3 weeks** or more.
* Your app **must comply** with all requirements in Google’s [[OAuth policies](https://developers.google.com/identity/protocols/oauth2)](https://developers.google.com/identity/protocols/oauth2).
* Without verification, you **cannot** access these scopes in production.

---

### 2. Integration Nightmare

According to the [[official documentation](https://developers.google.com/youtube/v3/docs/members/list)](https://developers.google.com/youtube/v3/docs/members/list), the endpoint is:

```
GET https://www.googleapis.com/youtube/v3/members
```

**Sounds simple?**
**No. Absolutely not.**

### ❗ Requirements:

* You **must be the OWNER** of the YouTube channel.
* The channel must be part of the **YouTube Partner Program**:
  ➡️ [[Program Requirements](https://support.google.com/youtube/answer/72851)](https://support.google.com/youtube/answer/72851)

### ✅ You must:

* Follow [[YouTube monetization policies](https://support.google.com/youtube/answer/1311392)](https://support.google.com/youtube/answer/1311392)
* Live in a [[supported country](https://support.google.com/youtube/answer/7101720)](https://support.google.com/youtube/answer/7101720)
* Have **no active** [[Community Guidelines strikes](https://support.google.com/youtube/answer/2802032)](https://support.google.com/youtube/answer/2802032)
* Enable [[2-Step Verification](https://myaccount.google.com/signinoptions/two-step-verification/enroll-welcome)](https://myaccount.google.com/signinoptions/two-step-verification/enroll-welcome)
* Have [[Advanced features](https://support.google.com/youtube/answer/9890437)](https://support.google.com/youtube/answer/9890437)
* Have an active [[AdSense for YouTube](https://support.google.com/youtube/answer/11602441)](https://support.google.com/youtube/answer/11602441) account linked

---

### 3. Reality Check: The 403 Error Wall

We’ve tried **everything** in every environment:

* Node.js (with official libraries)
* Vanilla JavaScript
* Postman
* Google Sheets Apps Script

**All attempts lead to the same result:**

```json
{
  "message": "Request failed with status code 403",
  "name": "AxiosError",
  "config": {
    "method": "get",
    "url": "https://www.googleapis.com/youtube/v3/members",
    "params": { "part": "snippet", "maxResults": 50 },
    "headers": {
      "Authorization": "Bearer _token",
      "Accept": "application/json"
    }
  },
  "code": "ERR_BAD_REQUEST",
  "status": 403
}
```

---

### 4. Google’s Official Response: **"Won’t Fix"**

In the earliest known thread about this issue (Feb 22, 2024), Google replied:

> **Status: Won't Fix (Intended Behavior)**
> "This means you don't have access to Channel Memberships API endpoint. It can be requested only through a **Partner Manager** or **Google Representative**."

Full context here:
🔗 [[Google Issue Tracker](https://issuetracker.google.com/issues/326362541?pli=1)](https://issuetracker.google.com/issues/326362541?pli=1)

### So we did what anyone would:

➡️ **Contacted YouTube Support** — They said:
*"Not our department, file an Issue Tracker."* → **Did that — no resolution.**

➡️ **Contacted OAuth Verification Team** — They said:
*"We can’t help with technical issues outside of OAuth verification."*
Standard reply pointing to these generic resources:

* [[General Support](https://support.google.com/googleapi/answer/7014572?hl=en)](https://support.google.com/googleapi/answer/7014572?hl=en)
* [[Stack Overflow](http://stackoverflow.com/questions/tagged/google-api)](http://stackoverflow.com/questions/tagged/google-api)

In short: **no one owns this problem** internally at Google.

---

### 5. Attempting Contact with a Google Representative

**Here’s the harsh truth:**

* There’s **no clear definition** or process to reach a "Google Representative" for this matter.
* We tried through the OAuth verification channel → got the usual "not our department" response.

**Their reply:**

> "*If users are receiving a 403 error, please see the [[support page](https://support.google.com/accounts/answer/12917337?hl=en)](https://support.google.com/accounts/answer/12917337?hl=en)... Unfortunately, the verification channel is not equipped to provide support on technical issues outside of OAuth Verification.*"

**No mention** of the Channel Memberships API or its restrictions.

---

### 6. Other Notable (but Useless) Resources

* **Stack Overflow thread:**
  [[Getting a 403 Forbidden error for YouTube Data API](https://stackoverflow.com/questions/51268669/getting-a-403-forbidden-error-for-youtube-data-api)](https://stackoverflow.com/questions/51268669/getting-a-403-forbidden-error-for-youtube-data-api)

* **Discord workaround?**
  There are **rumors** about using the Discord integration:
  [[Discord YouTube Channel Memberships FAQ](https://support.discord.com/hc/en-us/articles/215162978-YouTube-Channel-Memberships-Integration-FAQ)](https://support.discord.com/hc/en-us/articles/215162978-YouTube-Channel-Memberships-Integration-FAQ)
  But this is **only for linking Discord servers**, not for extracting members via API.

* **Another misleading thread:**
  [[Get a list of all members in a specific server](https://stackoverflow.com/questions/72851517/get-a-list-of-all-members-in-specific-server-automatically)](https://stackoverflow.com/questions/72851517/get-a-list-of-all-members-in-specific-server-automatically)
  Not related to YouTube Data API at all.

---

### ✅ The Only (Terrible) Workaround

Go to the [**[YouTube Data API Explorer](https://developers.google.com/youtube/v3/docs/members/list)**](https://developers.google.com/youtube/v3/docs/members/list):

1. Filter by **YouTube Data v3**.
2. Select `members > list`.
3. Click **“Try this API”**.
4. Use the **creator account** directly.

That’s it. No automation, no API keys, no backend systems — **manual testing only**.

---

### ❌ Conclusion: Not Possible

After **months of research**, including:

* All possible environments
* All official docs
* Direct contact with Google
* OAuth verification
* Testing multiple accounts

The only conclusion is:

> **There is NO public or automated way to retrieve YouTube channel members via the YouTube Data API (even if the docs saying that is possible and blá-blá) unless you’re a partner with a Google Representative or Manager (it's like a legend: no one ever seen or heard about them 😅), who can request access on your behalf.**

---

### 💡 Final Advice

* Don’t waste time trying to "solve" this.
* If this is critical to your project, consider alternative approaches, such as **integrating with Discord** or prompting users to **self-identify** via OAuth login.
* Keep monitoring [[Google Issue Tracker](https://issuetracker.google.com/issues/326362541?pli=1)](https://issuetracker.google.com/issues/326362541?pli=1) for any policy changes.

---

### 🔗 Relevant Links:

* [YouTube Data API `members.list`](https://developers.google.com/youtube/v3/docs/members/list)
* [[YouTube Partner Program Requirements](https://support.google.com/youtube/answer/72851)](https://support.google.com/youtube/answer/72851)
* [[OAuth Verification Process](https://support.google.com/cloud/answer/13463073)](https://support.google.com/cloud/answer/13463073)
* [[General Google API Support](https://support.google.com/googleapi/answer/7014572?hl=en)](https://support.google.com/googleapi/answer/7014572?hl=en)

---

Hope this helps save you **months** of frustration.
If anyone **actually** managed to get this working — **please share**!

We’d all be very interested to know.

**Comment by sonnar0486** [comment]
I found a workaround, and that's using StreamElements's API. Channels have to link their YouTube to StreamElements though, but they provide YouTube API access token which you can call `youtube/v3/members`

**Reply by jvmMachado** [reply]
Could you please tell me more about that? Do you mean the JWT Token they provide on profile page? I have tried that one with youtube API, but did not work. I searched through their docs on https://dev.streamelements.com/docs/api-docs/ but found no endpoint to get members list from. Did you have to use another authentication provided by StreamElements? OAuth 2.0 perhaps?

**Reply by sonnar0486** [reply]
Sorry it wasn't clear. You can call `https://api.streamelements.com/kappa/v2/channels/me` with the JWT Token and you can use the `accessToken` to call `https://www.googleapis.com/youtube/v3/members` since SE has the members API permission.

**Reply by jvmMachado** [reply]
Thank you for replying, mate. You really helped me a lot. I've been trying to get it to work for ages and your workaround is something I can really work with. Thank you very much!

**Reply by MjMthimunye** [reply]
@sonnar0486 Thanks for sharing this. It worked for me too. 
Very helpful.
